gbe/email/views/mail_to_bidders_view.py

```diff
@@ -75,26 +91,41 @@ def get_to_list(self):
             for bid in eval(bid_type).objects.filter(query):
                 bidder = (bid.profile.user_object.email,
                           bid.profile.display_name)
-                if bid.profile.email_allowed(
-                        self.email_type) and bidder not in to_list:
-                    to_list += [bidder]
+                if bid.profile.email_allowed(self.email_type) and (
+                        bidder not in any_list):
+                    if bidder not in exclude_list:
+                        any_list += [bidder]
+                    elif bidder not in already_excluded:
+                        already_excluded += [bidder]
             if draft:
                 for bid in eval(bid_type).objects.filter(draft_query):
                     bidder = (bid.profile.user_object.email,
                               bid.profile.display_name)
                     if bid.profile.email_allowed(self.email_type) and (
-                            bidder not in to_list):
-                        to_list += [bidder]
-        return sorted(to_list, key=lambda s: s[1].lower())
+                            bidder not in any_list):
+                        if bidder not in exclude_list:
+                            any_list += [bidder]
+                        elif bidder not in already_excluded:
+                            already_excluded += [bidder]
+        self.excluded_count = len(already_excluded)
+        return any_list
```

Comment:
I thought about conjoining this and the previous (identical) logic.  The problem is that it's a lot of list-passing around, which seems like it could easily be just as bug prone.